src/backend/command.ts

```diff
@@ -26,11 +35,21 @@ export interface Reversible<T extends Reversible<T>> {
  * which can be extended with attributes and constructors.
  */
 export abstract class CommandBase implements Command {
+  protected _requirements: Set<any> = new Set();
+
   public abstract execute(): Promise<void>;
 
   public then(next: Command): SequentialCommandGroup {
     return new SequentialCommandGroup([this, next]);
   }
+
+  public get requirements(): Set<any> {
+    return this._requirements;
+  }
+
+  public addRequirements(reqs: any[]) {

```

Comment:
@AlexKempen should addRequirements be added to the Command interface as well? I figured that since it's simply a helper method to add requirements to the protected Set, it can be up to any other implementation of Command that doesn't use CommandBase to figure out how to add requirements without forcing the method to every subclass.